File: atcoder/abc144D_water_bottle.py
# unihernandez22
# https://atcoder.jp/contests/abc144/tasks/abc144_d
# binary search, math

from math import *

a, b, x = map(int, input().split())

areatotal = a*a*b

if areatotal / 2 == x: print(45.0)
elif x > areatotal/2:
    left = 0.0
    right = 90.0
    while right-left > 0.00000001:
        current = (right+left)/2
        if a**2 * tan(radians(current)) / 2 * a < (areatotal - x):
            left = current
        else:
            right = current
    print(left)
else:
    left = 0.0
    right = 90.0
    while right-left > 0.00000001:
        current = (right+left)/2
        if b**2 * tan(radians(90-current))/2*a > x:
            left = current
        else:
            right = current
    print(left)

File: atcoder/abc144D_water_bottle.hs
-- unihernandez22
-- https://atcoder.jp/contests/abc144/tasks/abc144_d
-- math, trigonometry

-- hahahaha oneliner

main = do interact (\li -> (show $ (\x -> x / 6.283185307179586 * 360) $ ((\[a,b,x] -> atan(if x <= b/2 * (a^2) then ((a * b ^ 2) / ( 2 * x)) else ((2 * (b - (x / (a ^ 2)))) / a))) :: [Double] -> Double) $ ((map read $ words li) :: [Double])) ++ "\n")
